File: primeNumbers.cpp
#include <math.h>
#include <stdio.h>
#include <limits.h>

int main() {
  int n = 2000000;
  // int n = 120;
  int numbers[n];

  // range(2, n)
  int a = 0;
  int b = 2;
  while (b < n) {
    numbers[a] = b;
    a++;
    b++;
  }

  // prime numbers
  int i = 2;
  while (i <= sqrt(n)) {
    for (int j = 0; j < n; j++) {
      int currentNumber = numbers[j];
      if (-1 == currentNumber || i == currentNumber) {
        continue;
      }
      if ((currentNumber % i) == 0) {
        numbers[j] = -1;
      }
    }
    i++;
  }

  // print numbers
//  for (int i = 0; i < n; i++) {
//    if (-1 != numbers[i]) {
//      printf( "%d ", numbers[i]);
//    }
//  }

  return 0;
}

File: prime_numbers.php
<?php
$start = microtime(true);

$n = 2 * (10 ** 6);
//$n = 120;
$numbers = range(2, $n);

$i = 2;
while ($i <= sqrt($n)) {
    foreach ($numbers as $key => $number) {
        if ($number === $i) {
            continue;
        }
        if ($number % $i === 0) {
            unset($numbers[$key]);
        }
    }
    $i++;
}

//foreach ($numbers as $number) {
//    echo $number." ";
//}

echo PHP_EOL . PHP_EOL;
echo "time : ".(microtime(true) - $start)." secondes.".PHP_EOL;
echo "memory : ".((memory_get_usage()/1024)/1024)." Mo.";
